KubeArmor/core/karmorprobedata.go

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of KubeArmor
+
+package core
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+
+	cfg "github.com/kubearmor/KubeArmor/KubeArmor/config"
+)
+
+type KarmorData struct {
+	OSImage             string
+	KernelVersion       string
+	KubeletVersion      string
+	ContainerRuntime    string
+	ActiveLSM           string
+	KernelHeaderPresent bool
+	HostSecurity        bool
+	ContainerSecurity   bool
+	KubeArmorPosture    KubeArmorPostures
+}
+
+type KubeArmorPostures struct {
+	DefaultFilePosture             string
+	DefaultNetworkPosture          string
+	DefaultCapabilitiesPosture     string
+	HostDefaultFilePosture         string
+	HostDefaultNetworkPosture      string
+	HostDefaultCapabilitiesPosture string
+}
+
+func SetKarmorData(dm *KubeArmorDaemon) KarmorData {
+	var kd KarmorData
+	var postures KubeArmorPostures
+
+	postures.DefaultFilePosture = cfg.GlobalCfg.DefaultFilePosture
+	postures.DefaultNetworkPosture = cfg.GlobalCfg.DefaultNetworkPosture
+	postures.DefaultCapabilitiesPosture = cfg.GlobalCfg.DefaultCapabilitiesPosture
+	postures.HostDefaultFilePosture = cfg.GlobalCfg.HostDefaultFilePosture
+	postures.HostDefaultNetworkPosture = cfg.GlobalCfg.HostDefaultNetworkPosture
+	postures.HostDefaultCapabilitiesPosture = cfg.GlobalCfg.HostDefaultCapabilitiesPosture
+
+	kd.OSImage = dm.Node.OSImage
+	kd.ContainerRuntime = dm.Node.ContainerRuntimeVersion
+	kd.KernelVersion = dm.Node.KernelVersion
+	kd.KubeletVersion = dm.Node.KubeletVersion
+	kd.ContainerRuntime = dm.Node.ContainerRuntimeVersion
+	if dm.RuntimeEnforcer != nil {
+		kd.ActiveLSM = dm.RuntimeEnforcer.EnforcerType
+
+		if cfg.GlobalCfg.Policy {
+			kd.ContainerSecurity = true
+		}
+		if cfg.GlobalCfg.HostPolicy {
+			kd.HostSecurity = true
+		}
+	}
+	kd.KernelHeaderPresent = true //this is always true since kube armor is running
+	kd.KubeArmorPosture = postures
+	return kd
+}
+
+// StrToFile Function
+func KarmorDataToFile(val KarmorData, destFile string) {
```

Comment:
```suggestion
// WriteToFile writes given string to file as JSON
func WriteToFile(val interface{}, destFile string) error {
```

We should move this to `common` directory where other helper functions reside.